promq/prom/output.go

```
@@ -104,6 +105,46 @@ func ToPrettyFormat(res *promql.Result, outputType string, colorized bool) (*str
 			return nil, err
 		}
 		return o, nil
+	case "prometheus":
+		o, err := ToPrometheusFormat(res)
+		if err != nil {
+			return nil, err
+		}
+		return o, nil
 	}
+
 	return nil, fmt.Errorf("unsupported formatting option (%s)", outputType)
 }
+
+func ToPrometheusFormat(result *promql.Result) (*string, error) {
+	if result.Err != nil {
+		return nil, result.Err
+	}
+	if result.Warnings != nil {
+		for _, e := range result.Warnings {
+			fmt.Println(e)
+		}
+	}
+
+	vector, err := result.Vector()
+	if err != nil {
+		return proto.String(result.String()), nil
```

You don't want to also return the error here?

---

This is a bit confusing, the error is returned only when the result is not a `Vector` type. I changed it to a cast to make it more clear what's going on.

---

ah i see, maybe a comment would suffice then?

---

Sounds good, added a comment for clarity